### PR TITLE
Final Full stack group Capstone- Front-end

### PR DESCRIPTION
**Pull request by cosywasswa**
👇 In this group project, we implemented the following requirements.

- [ ] We called this app BookItNow for managing doctor reservations/appointments

- [ ] The user logs in to the website, only by typing the username

- [ ] In the navigation panel, the user can see links to:
"New Reservation" form.
"My reservations".
"Add doctor
"Delete doctor
"doctors
- [ ] On the main page, the user can see doctors that we selected as a theme

- [ ] When the user selects a specific doctor, they can see the with its full description

- [ ] In the details page, the user can click the "Reserve" button

- [ ] When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

- [ ]Created both mobile and desktop versions.

- [ ]  When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.

- [ ] When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list

- [ ] To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).

- [ ] The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.

- [ ] When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).

- [ ] Added full documentation for your API.

link to the Back end PR[click here](https://github.com/cosywasswa/BookItNow-back-end/pull/29)

### CODE REVIEW

File: src/redux/reservation/thunk.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// reservationThunks.js
```
Comment (Meltrust) [inline src/redux/reservation/thunk.js]: - [ ] Kindly, remove this piece of old commented code.  that way, the repo's cleanliness will be improved. 👍

File: src/redux/reservation/thunk.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import axios from 'axios';
+
+const url = 'https://bookitnow-kk0q.onrender.com/api/v1'; // Replace with your API endpoint
+
+export const fetchReservations = createAsyncThunk('reservations/fetchReservations', async (userId, thunkAPI) => {
+  try {
+    const response = await axios.get(`${url}/users/${userId}/reservations`);
+    //    console.log('Redux Store State:', response.data);
```
Comment (Meltrust) [inline src/redux/reservation/thunk.js]: - [ ] Kindly, remove this piece of old commented code.  that way, the repo's cleanliness will be improved. 👍

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import Doctors from './components/pages/doctors';
+import NewReservation from './components/pages/newReservation';
+import MyReservations from './components/pages/myReservations';
+import AddDoctor from './components/pages/addDoctor';
+import DeleteDoctor from './components/pages/deleteDoctor';
+import DoctorDetails from './components/doctor/doctorDetails';
+import Layout from './components/Nav/layout';
+import { useUser } from './components/userAccess/userContext';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js]: - [ ]  Please, kindly refactor this functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @cosywasswa,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:


-  Nice code organization ✔️ 
-  The app is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js (hunk 3 above)
Comment (cosywasswa) [reply src/App.js]: Thank you

**Review by Meltrust — APPROVED** [review]
#### Hi @cosywasswa,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Functions are now ES6✔️ 
- Commented code has been removed✔️ 
- The App is very beautiful ✔️ 
- The App is working well ✔️ 
- Superb job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**